demos/scalar_advection/scalar_advection.py

```diff
@@ -0,0 +1,107 @@
+# Demo for pure scalar advection - this is mostly copied from the Firedrake demo DG_advection,
+# but here we use G-ADOPT's Energy solver and use a CG discretisation
+# with Streamline Upwind (SU) stabilisation.
+
+from gadopt import *
+from gadopt.time_stepper import DIRK33
+
+# We use a 40-by-40 mesh of squares.
+mesh = UnitSquareMesh(40, 40, quadrilateral=True)
+
+# We set up a function space of bilinear elements for :math:`q`, and
+# a vector-valued continuous function space for our velocity field. ::
+
+V = FunctionSpace(mesh, "Q", 1)
+W = VectorFunctionSpace(mesh, "Q", 1)
```

Comment:
This is a little nitpicky, but we've so far used V and W for velocity and pressure spaces, and Q for the scalar in other demos

Reply:
done!